src/cascadia/TerminalApp/Jumplist.cpp

```diff
@@ -0,0 +1,142 @@
+    if (FAILED(_updateProfiles(jumplistItems, settings->GetProfiles())))
+    {
+        return;
+    }
+
+    // TODO: Add items from the future customizable new tab dropdown as well.
+    // This could either replace the default profiles, or be added alongside them.
+
+    // Add the items to the jumplist Task section.
+    // The Tasks section is immutable by the user, unlike the destinations
+    // section that can have its items pinned and removed.
+    jumplistInstance->AddUserTasks(jumplistItems.get());
+
+    if (FAILED(jumplistInstance->CommitList()))
+    {
+        return;
+    }
+}
+
+// Method Description:
+// - Creates and adds a ShellLink object to the Jumplist for each profile.
+// Arguments:
+// - jumplistItems - The jumplist item list
+// - profiles - The profiles to add to the jumplist
+// Return Value:
+// - S_OK or HRESULT failure code.
+HRESULT Jumplist::_updateProfiles(winrt::com_ptr<IObjectCollection>& jumplistItems, const gsl::span<const Profile>& profiles)
+{
+    HRESULT result = S_OK;
+
+    for (const auto& profile : profiles)
+    {
+        // Craft the arguments following "wt.exe"
+        auto profileName = profile.Name();
+        auto args = fmt::format(L"-p \"{}\"", profileName);
+
+        // Create the shell link object for the profile
+        winrt::com_ptr<IShellLink> shLink;
+        auto result = _createShellLink(profileName, profile.GetExpandedIconPath(), args, shLink);
+        if (FAILED(result))
+        {
+            return result;
+        }
+
+        jumplistItems->AddObject(shLink.get());
+    }
+
+    return result;
+}
+
+// Method Description:
+// - Creates a ShellLink object. Each item in a jumplist is a ShellLink, which is sort of
+//   like a shortcut. It requires the path to the application (wt.exe), the arguments to pass,
+//   and the path to the icon for the jumplist item. The path to the application isn't passed
+//   into this function, as we'll determine it with GetModuleFileName.
+// Arguments:
+// - name: The name of the item displayed in the jumplist.
+// - path: The path to the icon for the jumplist item.
+// - args: The arguments to pass along with wt.exe
+// Return Value:
+// - S_OK or HRESULT failure code.
+HRESULT Jumplist::_createShellLink(const std::wstring_view& name,
+                                   const std::wstring_view& path,
+                                   const std::wstring_view& args,
+                                   winrt::com_ptr<IShellLink>& shLink)
+{
+    // Create a shell link object
+    auto result = CoCreateInstance(CLSID_ShellLink, nullptr, CLSCTX_ALL, IID_PPV_ARGS(&shLink));
+    if (FAILED(result))
+    {
+        return result;
+    }
+
+    std::filesystem::path module{ wil::GetModuleFileNameW<std::wstring>(nullptr) };
+    shLink->SetPath(module.c_str());
+    shLink->SetArguments(args.data());
+
+    PROPVARIANT titleProp;
+    titleProp.vt = VT_LPWSTR;
+    titleProp.pwszVal = const_cast<wchar_t*>(name.data());

```

Comment:
This makes me cringe, but I guess it's the only way to manage it (and they're not going to modify it.)